test/unit/logic/transaction.js

```diff
@@ -0,0 +1,466 @@
+			validTransaction.asset = {
+				signature: {
+					publicKey: validSender.publicKey
+				}
+			};
+			transaction.verify(validTransaction, validSender, {}, function (err, res) {
+				expect(err).to.be.empty;
+				expect(res).to.be.empty;
+				done();
+			});
+		});
+
+		it.skip('should verify proper DAPP transaction with proper sender', function (done) {
+			validTransaction.type = transactionTypes.DAPP;
+			validTransaction.asset = {
+				signature: {
+					publicKey: validSender.publicKey
+				}
+			};
+			transaction.verify(validTransaction, validSender, {}, function (err, res) {
+				expect(err).to.be.empty;
+				expect(res).to.be.empty;
+				done();
+			});
+		});
+
+		it.skip('should verify proper IN_TRANSFER transaction with proper sender', function (done) {
+			validTransaction.type = transactionTypes.IN_TRANSFER;
+			validTransaction.asset = {
+				signature: {
+					publicKey: validSender.publicKey
+				}
+			};
+			transaction.verify(validTransaction, validSender, {}, function (err, res) {
+				expect(err).to.be.empty;
+				expect(res).to.be.empty;
+				done();
+			});
+		});
+
+		it.skip('should verify proper OUT_TRANSFER transaction with proper sender', function (done) {
+			validTransaction.type = transactionTypes.OUT_TRANSFER;
+			validTransaction.asset = {
+				signature: {
+					publicKey: validSender.publicKey
+				}
+			};
+			transaction.verify(validTransaction, validSender, {}, function (err, res) {
+				expect(err).to.be.empty;
+				expect(res).to.be.empty;
+				done();
+			});
+		});
+
+		it('should throw an error with no param', function () {
+			expect(transaction.verify).to.throw();
+		});
+	});
+
+	describe('verifySignature', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.verifySignature).to.throw();
+		});
+	});
+
+	describe('verifySecondSignature', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.verifySecondSignature).to.throw();
+		});
+	});
+
+	describe('verifyBytes', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.verifyBytes).to.throw();
+		});
+	});
+
+	describe('apply', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.apply).to.throw();
+		});
+	});
+
+	describe('undo', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.undo).to.throw();
+		});
+	});
+
+	describe('applyUnconfirmed', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.applyUnconfirmed).to.throw();
+		});
+	});
+
+	describe('undoUnconfirmed', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.undoUnconfirmed).to.throw();
+		});
+	});
+
+	describe('dbSave', function () {
+
+		it('should throw an error with no param', function () {
+			expect(transaction.dbSave).to.throw();
+		});
+	});
```

Comment:
Here we should test returned value includes optional data property. See: https://github.com/LiskHQ/lisk/blob/development/logic/transaction.js#L658

We also need to test: https://github.com/LiskHQ/lisk/blob/development/logic/transaction.js#L642